lib/web_ui/dev/test_runner.dart

```diff
@@ -159,22 +161,28 @@ class TestCommand extends Command<bool> with ArgUtils {
       await _runPubGet();
     }
 
-    // Many tabs will be left open after Safari runs, quit Safari during
-    // cleanup.
-    if (browser == 'safari') {
-      cleanupCallbacks.add(() async {
-        // Only close Safari if felt is running in CI environments. Do not close
-        // Safari for the local testing.
-        if (io.Platform.environment['LUCI_CONTEXT'] != null || isCirrus) {
-          print('INFO: Safari tests ran. Quit Safari.');
-          await runProcess(
-            'sudo',
-            ['pkill', '-lf', 'Safari'],
-            workingDirectory: environment.webUiRootDir.path,
-          );
-        } else {
-          print('INFO: Safari tests ran. Please quit Safari tabs.');
-        }
+    // In order to run iOS Safari unit tests we need to make sure, iOS
+    // Simulator is booted.
+    if (browser == 'ios-safari') {
+      final IosSimulatorManager iosSimulatorManager = IosSimulatorManager();
+      IosSimulator iosSimulator;
+      try {
+        iosSimulator = await iosSimulatorManager.getSimulator(
+            IosSafariArgParser.instance.iosMajorVersion,
+            IosSafariArgParser.instance.iosMinorVersion,
+            IosSafariArgParser.instance.iosDevice);
+      } catch (e) {
+        throw Exception('Error getting requested simulator. Consider using '
+            '`felt create` command instead. exception: $e');
+      }
+      if (!iosSimulator.booted) {
+        await iosSimulator.boot();
+        print('INFO: Simulator ${iosSimulator.id} booted.');
+      }
+
+      cleanupCallbacks.add(() {
+        iosSimulator.shutdown();
```

Comment:
Let's shutdown the simulator only if we started it. If we reused an already running simulator the user must have had it running for a reason and could be upset if we shut it down without asking.

Reply:
done.